components/Home/DevIncentives/requests.jsx

```diff
@@ -137,45 +147,81 @@ export const getMapUnitIncentivesRequest = async ({ unitType, unitId }) => {
 
-  // if the current epoch is the last epoch, calculate the incentives
-  if (currentEpochCounter === lastEpoch) {
-    const {
-      rewardUnitFraction: aRewardFraction,
-      topUpUnitFraction: aTopupFraction,
-    } = agentInfo;
-    const {
-      rewardUnitFraction: cRewardFraction,
-      topUpUnitFraction: cTopupFraction,
-    } = componentInfo;
-
-    /**
-     * for unitType agent(0) & component(1),
-     * the below formula is used to calculate the incentives
-     */
-    const componentPendingReward = (parseToEth(pendingRelativeReward) * cRewardFraction) / 100;
-    const agentPendingReward = (parseToEth(pendingRelativeReward) * aRewardFraction) / 100;
-    const componentPendingTopUp = (parseToEth(pendingRelativeTopUp) * cTopupFraction) / 100;
-    const agentPendingTopUp = (parseToEth(pendingRelativeTopUp) * aTopupFraction) / 100;
+  const rewardInEth = parseToEth(pendingRelativeReward);
+  const isCurrentEpochWithReward = currentEpochCounter === Number(lastEpoch) && rewardInEth > 0;
 
+  // if the current epoch is not the last epoch, return 0
+  if (!isCurrentEpochWithReward) {
     return {
-      pendingRelativeReward:
-        unitType === UNIT_TYPES.COMPONENT
-          ? componentPendingReward
-          : agentPendingReward,
-      pendingRelativeTopUp:
-        unitType === UNIT_TYPES.COMPONENT
-          ? componentPendingTopUp
-          : agentPendingTopUp,
+      pendingRelativeReward: 0,
+      pendingRelativeTopUp: 0,
       id: '0',
       key: '0',
     };
   }
 
+  // if the current epoch is the last epoch, calculate the incentives
+  const {
+    rewardUnitFraction: cRewardFraction,
+    topUpUnitFraction: cTopupFraction,
+    sumUnitTopUpsOLAS: cSumUnitTopUpsOLAS,
+  } = componentInfo;
+  const {
+    rewardUnitFraction: aRewardFraction,
+    topUpUnitFraction: aTopupFraction,
+    sumUnitTopUpsOLAS: aSumUnitTopUpsOLAS,
+  } = agentInfo;
+
+  /**
+   * for unitType agent(0) & component(1),
+   * the below calulcation is done to get the reward and topup
+   */
+  const componentReward = (rewardInEth * cRewardFraction) / 100;
+  const agentReward = (rewardInEth * aRewardFraction) / 100;
+
+  let totalIncentives = parseToEth(pendingRelativeTopUp);
+  let componentTopUp = 0;
+  let agentPendingTopUp = 0;
+
+  if (pendingRelativeTopUp > 0) {
+    const inflationPerEpoch = await contract.methods
+      .getInflationPerEpoch()
+      .call();
+    const inflationPerSecond = await contract.methods
+      .inflationPerSecond()
+      .call();
+    const epochLength = await getEpochLength();
+    const isValid = await canShowCheckpoint(true);
+
+    const totalTopUps = isValid
+      ? parseToEth(inflationPerSecond) * epochLength
+      : parseToEth(inflationPerEpoch);
+    totalIncentives *= totalTopUps;
+
+    const componentSumIncentivesInEth = parseToEth(cSumUnitTopUpsOLAS) * 100;
+    const agentSumIncentivesInEth = parseToEth(aSumUnitTopUpsOLAS) * 100;
```

Comment:
parseToEth loses a lot of precision. Please calculate all the numerators in `BigNumber` values, then parseToEth after the division in `BigNumber`.